sqlx-macros/src/derives/row.rs

```diff
@@ -66,8 +66,18 @@ fn expand_derive_from_row_struct(
     for field in fields {
         let ty = &field.ty;
 
-        predicates.push(parse_quote!(#ty: sqlx::decode::Decode<#lifetime, R::Database>));
-        predicates.push(parse_quote!(#ty: sqlx::types::Type<R::Database>));
+        let attributes = parse_child_attributes(&field.attrs).unwrap();
+
+        if attributes.flatten && attributes.default {
+            return Err(syn::Error::new_spanned(field, "sqlx: Field cannot be both 'default' and 'flatten'"));
```

Comment:
Nit: I would prefer `` `#[sqlx(default)]` and `#[sqlx(flatten)]` cannot be used on the same field``

Reply:
Actually, if we _do_ want to support this what we should do is allow `#[sqlx(default)]` on the struct itself which would change the emitted `FromRow` to do something like this:

```rust
let mut this: Self = core::default::Default::default();

if let Some(column) = row.get::<Option<_>>("<column name>") {
    this.field = column;
}

// etc.
```

and then the error above can recommend moving `#[sqlx(default)]` to the other struct instead.